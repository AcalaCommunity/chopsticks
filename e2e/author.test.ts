import { describe, expect, it } from 'vitest'

import { api, dev, expectJson, mockCallback, testingPairs } from './helper'

describe('author rpc', () => {
  it('works', async () => {
    const { alice, bob } = testingPairs()

    {
      const { callback, next } = mockCallback()
      await api.tx.balances.transfer(bob.address, 100).signAndSend(alice, callback)
      await dev.newBlock()

      await next()

      expect(callback.mock.calls).toMatchSnapshot()
      callback.mockClear()

      await expectJson(api.rpc.chain.getBlock()).toMatchSnapshot()
      await expectJson(api.query.system.account(alice.address)).toMatchSnapshot()
      await expectJson(api.query.system.account(bob.address)).toMatchSnapshot()
    }

    {
      const { callback, next } = mockCallback()
      await api.tx.balances.transfer(bob.address, 200).signAndSend(alice, callback)
      await dev.newBlock()

      await next()

      expect(callback.mock.calls).toMatchSnapshot()
      callback.mockClear()

      await expectJson(api.rpc.chain.getBlock()).toMatchSnapshot()
      await expectJson(api.query.system.account(alice.address)).toMatchSnapshot()
      await expectJson(api.query.system.account(bob.address)).toMatchSnapshot()
    }

    {
      const { callback, next } = mockCallback()
      await api.tx.balances.transfer(bob.address, 300).signAndSend(alice, callback)
      await dev.newBlock()

      await next()

      expect(callback.mock.calls).toMatchSnapshot()
      callback.mockClear()

      await expectJson(api.rpc.chain.getBlock()).toMatchSnapshot()
      await expectJson(api.query.system.account(alice.address)).toMatchSnapshot()
      await expectJson(api.query.system.account(bob.address)).toMatchSnapshot()
    }
  })
})
