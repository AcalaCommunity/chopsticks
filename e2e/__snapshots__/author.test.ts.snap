// Vitest Snapshot v1

exports[`author rpc > works 1`] = `
[
  [
    SubmittableResult {
      "dispatchError": undefined,
      "dispatchInfo": undefined,
      "events": [],
      "internalError": undefined,
      "status": {
        "ready": null,
      },
      "txHash": "0x379a84a326f2abe840fc3b54db762da86f60c31b20f7d81d15cc5ffb7dc995be",
      "txIndex": undefined,
    },
  ],
  [
    SubmittableResult {
      "dispatchError": undefined,
      "dispatchInfo": {
        "class": "Normal",
        "paysFee": "Yes",
        "weight": 166860000,
      },
      "events": [
        {
          "event": {
            "data": [
              "5FA9nQDVg267DEd8m1ZypXLBnvN7SFxYwV7ndqSYGiN9TTpu",
              15214802425,
            ],
            "index": "0x0a08",
          },
          "phase": {
            "applyExtrinsic": 2,
          },
          "topics": [],
        },
        {
          "event": {
            "data": [
              "5FA9nQDVg267DEd8m1ZypXLBnvN7SFxYwV7ndqSYGiN9TTpu",
              "5GoNkf6WdbxCFnPdAnYYQyCjAKPJgLNxXwPjwTh6DGg6gN3E",
              100,
            ],
            "index": "0x0a02",
          },
          "phase": {
            "applyExtrinsic": 2,
          },
          "topics": [],
        },
        {
          "event": {
            "data": [
              "5EYCAe5fiQJsoL71D8qNrZ7NznTuLKXk24Wjp9pe18su2Yk1",
              12171841940,
            ],
            "index": "0x0a07",
          },
          "phase": {
            "applyExtrinsic": 2,
          },
          "topics": [],
        },
        {
          "event": {
            "data": [
              12171841940,
            ],
            "index": "0x1406",
          },
          "phase": {
            "applyExtrinsic": 2,
          },
          "topics": [],
        },
        {
          "event": {
            "data": [
              "5EYCAe5fiQJsnou3Fd4QznxPAJXfuj3yDpKUnabchMqHdBoQ",
              3042960485,
            ],
            "index": "0x0a07",
          },
          "phase": {
            "applyExtrinsic": 2,
          },
          "topics": [],
        },
        {
          "event": {
            "data": [
              "5FA9nQDVg267DEd8m1ZypXLBnvN7SFxYwV7ndqSYGiN9TTpu",
              15214802425,
              0,
              0,
            ],
            "index": "0x0e03",
          },
          "phase": {
            "applyExtrinsic": 2,
          },
          "topics": [],
        },
        {
          "event": {
            "data": [
              {
                "class": "Normal",
                "paysFee": "Yes",
                "weight": 166860000,
              },
            ],
            "index": "0x0000",
          },
          "phase": {
            "applyExtrinsic": 2,
          },
          "topics": [],
        },
      ],
      "internalError": undefined,
      "status": {
        "finalized": "0xeefde1ec4bb9834c9282bdaa2034a1f603ef931b70c25caaa5a97f1fb2720fd5",
      },
      "txHash": "0x379a84a326f2abe840fc3b54db762da86f60c31b20f7d81d15cc5ffb7dc995be",
      "txIndex": 2,
    },
  ],
]
`;

exports[`author rpc > works 2`] = `
{
  "block": {
    "extrinsics": [
      "0x280401000b207230268401",
      "0xa91704a000e902d94d2e18b4a8a88d9b21cb03bbe4df4431a93718c2c602c13043d595e8b8735a3e910b0021c0ecb97a4c7cc86aa22eeb2db02b60829551cf67bdc490f0948ac009f6c0801257ffee8b177dab50e7b537ccbd2d1ffcd8a0068a08d57a92ce367e4e0f52e1080661757261209aa0470800000000056175726101013aefb1326d057c6d6d174f1565404a8b1a6ab45c941fa89bd8a56ac401226d1b0d96a6e436d7021dd097d316f07fe554fc35c92e1966e8a9b6ebc4de15e57886901e07009c325649c4ebff345980ea29b5fb7b60a9af1f8cb26e5d9403418e89aee6d5c80000500034c45e414cb008e0e61e46722aa60abdd67280bd82018ee510cf360e1a460326ef53dde570b25b9e835fc045551ffe4e88fe08c45ee678799d3eff024253b90e84927cc6807ed4bd40e4853d40a1fd44e7128aa75ded4660543dc48918f507ac56fbef7d97cc5f0077dfdb8adb10f78f10a5df8742c5458401df75b8601e1b21393fc7772afc0788aa25e88713977d61495fabb2550af5c83545035f04b49d95320d9021994c850f25b8e385f902000030000080000008000000000010000000100005000000050000000a0000000a000000000050000000100000e8764817000000040000000400000000000000000000000000000000000000000000000000000000000000000000000800000000200000040000000400000000001000b0040000000000000000000014000000040000000400000000000000010100000000060000006400000002000000c8000000020000001900000000000000020000000200000000c817a804000000150180001480451c10fa62e420ae9678ed9e9f7f19e3e39be3a38ed4e3fe314234f3e6fed31780d3dac5ff6c2ce4154f56c20524e2dd095001f40b34d7c8afca34011128bc86201501800804809f3ada68c357b5e0a3ebb39ef181acfa9943af4725c244330a4b2c60837612e88097425c486ee3f57ab095f94c7dc030d79c63147ff57c50c90a577e2ad727d2e2a10280439080216c4ed723e0a878bd96314428b2b7cb9727481f149f84011c0405423bb94e26805bd412b85f2239198146f1fe278f482ed1e54fff8aab83ff5278486f27553e5180118f288193e4da2abce3289ebcfe1b730737e817b189512053eb437b1211c9f6807a4cfe27b7daddf656ec94d5b0d02fa22183c8bcdadf8364f236cbe956cc8f9e80c3485bf9a1e0e78ac17578cca0d0b0cd5f632bd925284c5f54decc00e6f44aae6d019e0da05ca59913bc38a8630590f2627c14404c5f0a351b6a99a5b21324516e668bb86a570400505f0e7b9012096b41c4eb3aaf947f6ea4290800007c7700e67da63472835bb0b737093a19ad4c63f5a4efb16ffa83d007000004001d019ef78c98723ddc9073523ef3beefda0c1004505f0e7b9012096b41c4eb3aaf947f6ea4290800007c77095dac46c07a40d91506e7637ec4ba5763f5a4efb16ffa83d00700000400f09f05207f03cfdce586301014700e2c25931040505f0e7b9012096b41c4eb3aaf947f6ea4290800004c5f0ec2d17a76153ff51817f12d9cfc3c7f040025019f06de3d8a54d27e44a9d5ce189618f22d1008505f0e7b9012096b41c4eb3aaf947f6ea42908010080c74756edffa217dfb07ab596d82753deff985ac215e5cc2997d29afe1d397c16ed029f0cb6f36e027abb2091cfb5110ab5087f9960685f06155b3cd9a8c9e5e9a23fd5dc13a5ed2037418f1000000000685f08316cbf8fa0da822a20ac1c55bf1be3208212000000000000505f0e7b9012096b41c4eb3aaf947f6ea42908000080740bfbe51a080a46d5c79a4b185680b175abb3ccdb602dbdba7552f7172ed8c180fddafc590530ce9037825fac38ceee0615edf7def5fd0af0bacb16f7d47f9859685f090e2fbf2d792cb324bffa9427fe1f0e202a1e07008e1e0700c1019f0d710b30bd2eab0352ddcc26417aa19454807c77081e0bfde17b36573208a06cb5cfba6b63f5a4efb16ffa83d00700000402505f0e7b9012096b41c4eb3aaf947f6ea4290800004c5f03c716fb8fff3de61a883bb76adb34a204004c5f0dad6eef5c4b1c68eaa71ea17a02d9de04000000",
      "0x3102840088dc3417d5058ec4b4503e0c12ea1a0a89be200fe98922423d4334014fa6b0ee00f7b92e81c4c3678cf2bd0ce0774782709b8792d46abcb381c5617fd79550a7b3a0d0cd858419dbadbb4e763960b710e2265e0d21bf33ba1009184859d9618d06040100000a0000d17c2d7823ebf260fd138f2d7e27d114c0145d968b5ff5006125f2414fadae699101",
    ],
    "header": {
      "digest": {
        "logs": [
          {
            "preRuntime": [
              "0x61757261",
              "0xe1cf470800000000",
            ],
          },
          {
            "seal": [
              "0x61757261",
              "0x46e076a482debb2fe5d839f642f585570e9178620d731de559515666b95dda10eb4736d397e98b64afe66f2dbf7e1a3a6eeb394aa2cf9b5317e7dfcff0614e8f",
            ],
          },
        ],
      },
      "extrinsicsRoot": "0x0000000000000000000000000000000000000000000000000000000000000000",
      "number": 189521,
      "parentHash": "0x062327512615cd62ea8c57652a04a6c937b112f1410520d83e2fafb9776cdbe1",
      "stateRoot": "0x0000000000000000000000000000000000000000000000000000000000000000",
    },
  },
  "justifications": null,
}
`;

exports[`author rpc > works 3`] = `
{
  "consumers": 0,
  "data": {
    "feeFrozen": 0,
    "free": 4984785197475,
    "miscFrozen": 0,
    "reserved": 0,
  },
  "nonce": 1,
  "providers": 1,
  "sufficients": 0,
}
`;

exports[`author rpc > works 4`] = `
{
  "consumers": 0,
  "data": {
    "feeFrozen": 0,
    "free": 1000000000100,
    "miscFrozen": 0,
    "reserved": 0,
  },
  "nonce": 0,
  "providers": 1,
  "sufficients": 0,
}
`;

exports[`author rpc > works 5`] = `
[
  [
    SubmittableResult {
      "dispatchError": undefined,
      "dispatchInfo": undefined,
      "events": [],
      "internalError": undefined,
      "status": {
        "ready": null,
      },
      "txHash": "0x7a50dcf54f3f69a233ef03fd835b26b414c2892ccc0aad4f5b1a7b7407dbb16b",
      "txIndex": undefined,
    },
  ],
  [
    SubmittableResult {
      "dispatchError": undefined,
      "dispatchInfo": {
        "class": "Normal",
        "paysFee": "Yes",
        "weight": 166860000,
      },
      "events": [
        {
          "event": {
            "data": [
              "5FA9nQDVg267DEd8m1ZypXLBnvN7SFxYwV7ndqSYGiN9TTpu",
              15214802389,
            ],
            "index": "0x0a08",
          },
          "phase": {
            "applyExtrinsic": 2,
          },
          "topics": [],
        },
        {
          "event": {
            "data": [
              "5FA9nQDVg267DEd8m1ZypXLBnvN7SFxYwV7ndqSYGiN9TTpu",
              "5GoNkf6WdbxCFnPdAnYYQyCjAKPJgLNxXwPjwTh6DGg6gN3E",
              200,
            ],
            "index": "0x0a02",
          },
          "phase": {
            "applyExtrinsic": 2,
          },
          "topics": [],
        },
        {
          "event": {
            "data": [
              "5EYCAe5fiQJsoL71D8qNrZ7NznTuLKXk24Wjp9pe18su2Yk1",
              12171841911,
            ],
            "index": "0x0a07",
          },
          "phase": {
            "applyExtrinsic": 2,
          },
          "topics": [],
        },
        {
          "event": {
            "data": [
              12171841911,
            ],
            "index": "0x1406",
          },
          "phase": {
            "applyExtrinsic": 2,
          },
          "topics": [],
        },
        {
          "event": {
            "data": [
              "5EYCAe5fiQJsnou3Fd4QznxPAJXfuj3yDpKUnabchMqHdBoQ",
              3042960478,
            ],
            "index": "0x0a07",
          },
          "phase": {
            "applyExtrinsic": 2,
          },
          "topics": [],
        },
        {
          "event": {
            "data": [
              "5FA9nQDVg267DEd8m1ZypXLBnvN7SFxYwV7ndqSYGiN9TTpu",
              15214802389,
              0,
              0,
            ],
            "index": "0x0e03",
          },
          "phase": {
            "applyExtrinsic": 2,
          },
          "topics": [],
        },
        {
          "event": {
            "data": [
              {
                "class": "Normal",
                "paysFee": "Yes",
                "weight": 166860000,
              },
            ],
            "index": "0x0000",
          },
          "phase": {
            "applyExtrinsic": 2,
          },
          "topics": [],
        },
      ],
      "internalError": undefined,
      "status": {
        "finalized": "0x0a88e86168da4aa27a33f9273b34bba3dbef86bd7b246e831333c83a8d38f975",
      },
      "txHash": "0x7a50dcf54f3f69a233ef03fd835b26b414c2892ccc0aad4f5b1a7b7407dbb16b",
      "txIndex": 2,
    },
  ],
]
`;

exports[`author rpc > works 6`] = `
{
  "block": {
    "extrinsics": [
      "0x280401000b40c030268401",
      "0xa91704a000e902d94d2e18b4a8a88d9b21cb03bbe4df4431a93718c2c602c13043d595e8b8735a3e910b0021c0ecb97a4c7cc86aa22eeb2db02b60829551cf67bdc490f0948ac009f6c0801257ffee8b177dab50e7b537ccbd2d1ffcd8a0068a08d57a92ce367e4e0f52e1080661757261209aa0470800000000056175726101013aefb1326d057c6d6d174f1565404a8b1a6ab45c941fa89bd8a56ac401226d1b0d96a6e436d7021dd097d316f07fe554fc35c92e1966e8a9b6ebc4de15e57886921e07009c325649c4ebff345980ea29b5fb7b60a9af1f8cb26e5d9403418e89aee6d5c80000500034c45e414cb008e0e61e46722aa60abdd67280bd82018ee510cf360e1a460326ef53dde570b25b9e835fc045551ffe4e88fe08c45ee678799d3eff024253b90e84927cc6807ed4bd40e4853d40a1fd44e7128aa75ded4660543dc48918f507ac56fbef7d97cc5f0077dfdb8adb10f78f10a5df8742c5458401df75b8601e1b21393fc7772afc0788aa25e88713977d61495fabb2550af5c83545035f04b49d95320d9021994c850f25b8e385f902000030000080000008000000000010000000100005000000050000000a0000000a000000000050000000100000e8764817000000040000000400000000000000000000000000000000000000000000000000000000000000000000000800000000200000040000000400000000001000b0040000000000000000000014000000040000000400000000000000010100000000060000006400000002000000c8000000020000001900000000000000020000000200000000c817a804000000150180001480451c10fa62e420ae9678ed9e9f7f19e3e39be3a38ed4e3fe314234f3e6fed31780d3dac5ff6c2ce4154f56c20524e2dd095001f40b34d7c8afca34011128bc86201501800804809f3ada68c357b5e0a3ebb39ef181acfa9943af4725c244330a4b2c60837612e88097425c486ee3f57ab095f94c7dc030d79c63147ff57c50c90a577e2ad727d2e2a10280439080216c4ed723e0a878bd96314428b2b7cb9727481f149f84011c0405423bb94e26805bd412b85f2239198146f1fe278f482ed1e54fff8aab83ff5278486f27553e5180118f288193e4da2abce3289ebcfe1b730737e817b189512053eb437b1211c9f6807a4cfe27b7daddf656ec94d5b0d02fa22183c8bcdadf8364f236cbe956cc8f9e80c3485bf9a1e0e78ac17578cca0d0b0cd5f632bd925284c5f54decc00e6f44aae6d019e0da05ca59913bc38a8630590f2627c14404c5f0a351b6a99a5b21324516e668bb86a570400505f0e7b9012096b41c4eb3aaf947f6ea4290800007c7700e67da63472835bb0b737093a19ad4c63f5a4efb16ffa83d007000004001d019ef78c98723ddc9073523ef3beefda0c1004505f0e7b9012096b41c4eb3aaf947f6ea4290800007c77095dac46c07a40d91506e7637ec4ba5763f5a4efb16ffa83d00700000400f09f05207f03cfdce586301014700e2c25931040505f0e7b9012096b41c4eb3aaf947f6ea4290800004c5f0ec2d17a76153ff51817f12d9cfc3c7f040025019f06de3d8a54d27e44a9d5ce189618f22d1008505f0e7b9012096b41c4eb3aaf947f6ea42908010080c74756edffa217dfb07ab596d82753deff985ac215e5cc2997d29afe1d397c16ed029f0cb6f36e027abb2091cfb5110ab5087f9960685f06155b3cd9a8c9e5e9a23fd5dc13a5ed2037418f1000000000685f08316cbf8fa0da822a20ac1c55bf1be3208212000000000000505f0e7b9012096b41c4eb3aaf947f6ea42908000080740bfbe51a080a46d5c79a4b185680b175abb3ccdb602dbdba7552f7172ed8c180fddafc590530ce9037825fac38ceee0615edf7def5fd0af0bacb16f7d47f9859685f090e2fbf2d792cb324bffa9427fe1f0e202a1e07008e1e0700c1019f0d710b30bd2eab0352ddcc26417aa19454807c77081e0bfde17b36573208a06cb5cfba6b63f5a4efb16ffa83d00700000402505f0e7b9012096b41c4eb3aaf947f6ea4290800004c5f03c716fb8fff3de61a883bb76adb34a204004c5f0dad6eef5c4b1c68eaa71ea17a02d9de04000000",
      "0x3102840088dc3417d5058ec4b4503e0c12ea1a0a89be200fe98922423d4334014fa6b0ee00ac0122165c15db54e67d09a8f7aa0651c39c80bc775483f15a06dab12f7214019d6ae60ef92f4218819d4ddd2e2726bf500e82e405de7306aecbd6959701ef08140104000a0000d17c2d7823ebf260fd138f2d7e27d114c0145d968b5ff5006125f2414fadae692103",
    ],
    "header": {
      "digest": {
        "logs": [
          {
            "preRuntime": [
              "0x61757261",
              "0xe3cf470800000000",
            ],
          },
          {
            "seal": [
              "0x61757261",
              "0x46e076a482debb2fe5d839f642f585570e9178620d731de559515666b95dda10eb4736d397e98b64afe66f2dbf7e1a3a6eeb394aa2cf9b5317e7dfcff0614e8f",
            ],
          },
        ],
      },
      "extrinsicsRoot": "0x0000000000000000000000000000000000000000000000000000000000000000",
      "number": 189522,
      "parentHash": "0xeefde1ec4bb9834c9282bdaa2034a1f603ef931b70c25caaa5a97f1fb2720fd5",
      "stateRoot": "0x0000000000000000000000000000000000000000000000000000000000000000",
    },
  },
  "justifications": null,
}
`;

exports[`author rpc > works 7`] = `
{
  "consumers": 0,
  "data": {
    "feeFrozen": 0,
    "free": 4969570394886,
    "miscFrozen": 0,
    "reserved": 0,
  },
  "nonce": 2,
  "providers": 1,
  "sufficients": 0,
}
`;

exports[`author rpc > works 8`] = `
{
  "consumers": 0,
  "data": {
    "feeFrozen": 0,
    "free": 1000000000300,
    "miscFrozen": 0,
    "reserved": 0,
  },
  "nonce": 0,
  "providers": 1,
  "sufficients": 0,
}
`;

exports[`author rpc > works 9`] = `
[
  [
    SubmittableResult {
      "dispatchError": undefined,
      "dispatchInfo": undefined,
      "events": [],
      "internalError": undefined,
      "status": {
        "ready": null,
      },
      "txHash": "0xa250b7e868080aab25980d209bfd59c5ca9fdc98391820d56ea50454901ac6c3",
      "txIndex": undefined,
    },
  ],
  [
    SubmittableResult {
      "dispatchError": undefined,
      "dispatchInfo": {
        "class": "Normal",
        "paysFee": "Yes",
        "weight": 166860000,
      },
      "events": [
        {
          "event": {
            "data": [
              "5FA9nQDVg267DEd8m1ZypXLBnvN7SFxYwV7ndqSYGiN9TTpu",
              15214802352,
            ],
            "index": "0x0a08",
          },
          "phase": {
            "applyExtrinsic": 2,
          },
          "topics": [],
        },
        {
          "event": {
            "data": [
              "5FA9nQDVg267DEd8m1ZypXLBnvN7SFxYwV7ndqSYGiN9TTpu",
              "5GoNkf6WdbxCFnPdAnYYQyCjAKPJgLNxXwPjwTh6DGg6gN3E",
              300,
            ],
            "index": "0x0a02",
          },
          "phase": {
            "applyExtrinsic": 2,
          },
          "topics": [],
        },
        {
          "event": {
            "data": [
              "5EYCAe5fiQJsoL71D8qNrZ7NznTuLKXk24Wjp9pe18su2Yk1",
              12171841881,
            ],
            "index": "0x0a07",
          },
          "phase": {
            "applyExtrinsic": 2,
          },
          "topics": [],
        },
        {
          "event": {
            "data": [
              12171841881,
            ],
            "index": "0x1406",
          },
          "phase": {
            "applyExtrinsic": 2,
          },
          "topics": [],
        },
        {
          "event": {
            "data": [
              "5EYCAe5fiQJsnou3Fd4QznxPAJXfuj3yDpKUnabchMqHdBoQ",
              3042960471,
            ],
            "index": "0x0a07",
          },
          "phase": {
            "applyExtrinsic": 2,
          },
          "topics": [],
        },
        {
          "event": {
            "data": [
              "5FA9nQDVg267DEd8m1ZypXLBnvN7SFxYwV7ndqSYGiN9TTpu",
              15214802352,
              0,
              0,
            ],
            "index": "0x0e03",
          },
          "phase": {
            "applyExtrinsic": 2,
          },
          "topics": [],
        },
        {
          "event": {
            "data": [
              {
                "class": "Normal",
                "paysFee": "Yes",
                "weight": 166860000,
              },
            ],
            "index": "0x0000",
          },
          "phase": {
            "applyExtrinsic": 2,
          },
          "topics": [],
        },
      ],
      "internalError": undefined,
      "status": {
        "finalized": "0xeaf40e1042a99e9c20fa0b03d0806ce738885597e5ef204dbdf6f325ad905c9e",
      },
      "txHash": "0xa250b7e868080aab25980d209bfd59c5ca9fdc98391820d56ea50454901ac6c3",
      "txIndex": 2,
    },
  ],
]
`;

exports[`author rpc > works 10`] = `
{
  "block": {
    "extrinsics": [
      "0x280401000b600e31268401",
      "0xa91704a000e902d94d2e18b4a8a88d9b21cb03bbe4df4431a93718c2c602c13043d595e8b8735a3e910b0021c0ecb97a4c7cc86aa22eeb2db02b60829551cf67bdc490f0948ac009f6c0801257ffee8b177dab50e7b537ccbd2d1ffcd8a0068a08d57a92ce367e4e0f52e1080661757261209aa0470800000000056175726101013aefb1326d057c6d6d174f1565404a8b1a6ab45c941fa89bd8a56ac401226d1b0d96a6e436d7021dd097d316f07fe554fc35c92e1966e8a9b6ebc4de15e57886941e07009c325649c4ebff345980ea29b5fb7b60a9af1f8cb26e5d9403418e89aee6d5c80000500034c45e414cb008e0e61e46722aa60abdd67280bd82018ee510cf360e1a460326ef53dde570b25b9e835fc045551ffe4e88fe08c45ee678799d3eff024253b90e84927cc6807ed4bd40e4853d40a1fd44e7128aa75ded4660543dc48918f507ac56fbef7d97cc5f0077dfdb8adb10f78f10a5df8742c5458401df75b8601e1b21393fc7772afc0788aa25e88713977d61495fabb2550af5c83545035f04b49d95320d9021994c850f25b8e385f902000030000080000008000000000010000000100005000000050000000a0000000a000000000050000000100000e8764817000000040000000400000000000000000000000000000000000000000000000000000000000000000000000800000000200000040000000400000000001000b0040000000000000000000014000000040000000400000000000000010100000000060000006400000002000000c8000000020000001900000000000000020000000200000000c817a804000000150180001480451c10fa62e420ae9678ed9e9f7f19e3e39be3a38ed4e3fe314234f3e6fed31780d3dac5ff6c2ce4154f56c20524e2dd095001f40b34d7c8afca34011128bc86201501800804809f3ada68c357b5e0a3ebb39ef181acfa9943af4725c244330a4b2c60837612e88097425c486ee3f57ab095f94c7dc030d79c63147ff57c50c90a577e2ad727d2e2a10280439080216c4ed723e0a878bd96314428b2b7cb9727481f149f84011c0405423bb94e26805bd412b85f2239198146f1fe278f482ed1e54fff8aab83ff5278486f27553e5180118f288193e4da2abce3289ebcfe1b730737e817b189512053eb437b1211c9f6807a4cfe27b7daddf656ec94d5b0d02fa22183c8bcdadf8364f236cbe956cc8f9e80c3485bf9a1e0e78ac17578cca0d0b0cd5f632bd925284c5f54decc00e6f44aae6d019e0da05ca59913bc38a8630590f2627c14404c5f0a351b6a99a5b21324516e668bb86a570400505f0e7b9012096b41c4eb3aaf947f6ea4290800007c7700e67da63472835bb0b737093a19ad4c63f5a4efb16ffa83d007000004001d019ef78c98723ddc9073523ef3beefda0c1004505f0e7b9012096b41c4eb3aaf947f6ea4290800007c77095dac46c07a40d91506e7637ec4ba5763f5a4efb16ffa83d00700000400f09f05207f03cfdce586301014700e2c25931040505f0e7b9012096b41c4eb3aaf947f6ea4290800004c5f0ec2d17a76153ff51817f12d9cfc3c7f040025019f06de3d8a54d27e44a9d5ce189618f22d1008505f0e7b9012096b41c4eb3aaf947f6ea42908010080c74756edffa217dfb07ab596d82753deff985ac215e5cc2997d29afe1d397c16ed029f0cb6f36e027abb2091cfb5110ab5087f9960685f06155b3cd9a8c9e5e9a23fd5dc13a5ed2037418f1000000000685f08316cbf8fa0da822a20ac1c55bf1be3208212000000000000505f0e7b9012096b41c4eb3aaf947f6ea42908000080740bfbe51a080a46d5c79a4b185680b175abb3ccdb602dbdba7552f7172ed8c180fddafc590530ce9037825fac38ceee0615edf7def5fd0af0bacb16f7d47f9859685f090e2fbf2d792cb324bffa9427fe1f0e202a1e07008e1e0700c1019f0d710b30bd2eab0352ddcc26417aa19454807c77081e0bfde17b36573208a06cb5cfba6b63f5a4efb16ffa83d00700000402505f0e7b9012096b41c4eb3aaf947f6ea4290800004c5f03c716fb8fff3de61a883bb76adb34a204004c5f0dad6eef5c4b1c68eaa71ea17a02d9de04000000",
      "0x3102840088dc3417d5058ec4b4503e0c12ea1a0a89be200fe98922423d4334014fa6b0ee007aa899b75e856b8caa34caec04857abe9f7d921b1e92e6867f7e906211d0f9e977d10c4d26e1a254708e77168f506e9a711369993025ccb379dc695f935b540f240108000a0000d17c2d7823ebf260fd138f2d7e27d114c0145d968b5ff5006125f2414fadae69b104",
    ],
    "header": {
      "digest": {
        "logs": [
          {
            "preRuntime": [
              "0x61757261",
              "0xe5cf470800000000",
            ],
          },
          {
            "seal": [
              "0x61757261",
              "0x46e076a482debb2fe5d839f642f585570e9178620d731de559515666b95dda10eb4736d397e98b64afe66f2dbf7e1a3a6eeb394aa2cf9b5317e7dfcff0614e8f",
            ],
          },
        ],
      },
      "extrinsicsRoot": "0x0000000000000000000000000000000000000000000000000000000000000000",
      "number": 189523,
      "parentHash": "0x0a88e86168da4aa27a33f9273b34bba3dbef86bd7b246e831333c83a8d38f975",
      "stateRoot": "0x0000000000000000000000000000000000000000000000000000000000000000",
    },
  },
  "justifications": null,
}
`;

exports[`author rpc > works 11`] = `
{
  "consumers": 0,
  "data": {
    "feeFrozen": 0,
    "free": 4954355592234,
    "miscFrozen": 0,
    "reserved": 0,
  },
  "nonce": 3,
  "providers": 1,
  "sufficients": 0,
}
`;

exports[`author rpc > works 12`] = `
{
  "consumers": 0,
  "data": {
    "feeFrozen": 0,
    "free": 1000000000600,
    "miscFrozen": 0,
    "reserved": 0,
  },
  "nonce": 0,
  "providers": 1,
  "sufficients": 0,
}
`;
